src/internal/utils/node-belongs.ts

```diff
@@ -17,8 +17,12 @@ export function nodeBelongs(container: Node | null, target: Node | EventTarget |
   }
   const portal = findUpUntil(
     target as HTMLElement,
-    node => node instanceof HTMLElement && !!node.dataset.awsuiReferrerId
+    node => node === container || (node instanceof HTMLElement && !!node.dataset.awsuiReferrerId)
   );
+  if (portal === container) {
```

Comment:
Feels unlikely, but we should probably check here that we actually found a portal (`portal && portal === container`). Otherwise there could be a weird edge case where `nodeBelongs` returns true for `container==null` when the target is not inside a portal.